Source/Core/Check.js

```diff
@@ -0,0 +1,144 @@
+/*global define*/
+define([
+    './defaultValue',
+    './defined',
+    './DeveloperError',
+    './isArray'
+    ], function(
+        defaultValue,
+        defined,
+        DeveloperError,
+        isArray) {
+    'use strict';
+
+    /**
+     * Contains functions for checking that supplied arguments are of a specified type
+     * or meet specified conditions
+     * @private
+     *
+     * @exports Check
+     */
+    var Check = {};
+
+    Check.typeOf = {};
+    Check.numeric = {};
+
+    var errors = {
+        defined: getUndefinedErrorMessage,
+        failedType: getFailedTypeErrorMessage
+    };
+
+    function getUndefinedErrorMessage(name) {
+        return defaultValue(name, 'a value') + ' is required but is undefined.';
```

Comment:
I rarely run into thrown `DeveloperError`s when coding (not to say this is true for all users in general), but we always want to provide the most precise information possible so I think we are doing the users a disservice by giving them a more vague error message when we have the data readily available to be more precise.